util/testutil/integration/util_unix.go

```diff
@@ -0,0 +1,40 @@
+//go:build !windows
+// +build !windows
+
+package integration
+
+import (
+	"net"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func waitSocket(address string, d time.Duration, cmd *exec.Cmd) error {
+	address = strings.TrimPrefix(address, "unix://")
+	addr, err := net.ResolveUnixAddr("unix", address)
+	if err != nil {
+		return errors.Wrapf(err, "failed resolving unix addr: %s", address)
+	}
+
+	step := 50 * time.Millisecond
```

Comment:
Looks like this could be optimized so that the retry logic is the same for all arch.

Reply:
You suggest we abstract this in one common function for all archs?

Reply:
refactored in the coming commit.